src/test/java/com/fasterxml/jackson/databind/deser/jdk/StringArrayDeserializer5165Test.java

```diff
@@ -0,0 +1,88 @@
+package com.fasterxml.jackson.databind.deser.jdk;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.Nulls;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidNullException;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+// For [databind#5165]
+public class StringArrayDeserializer5165Test
+{
+    static class Dst {
+        private String[] array;
+
+        public String[] getArray() {
+            return array;
+        }
+
+        public void setArray(String[] array) {
+            this.array = array;
+        }
+    }
+
+    // Custom deserializer that converts empty strings to null
+    static class EmptyStringToNullDeserializer extends StdDeserializer<String> {
+        private static final long serialVersionUID = 1L;
+
+        public EmptyStringToNullDeserializer() {
+            super(String.class);
+        }
+
+        @Override
+        public String deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+            String value = p.getValueAsString();
+            if (value != null && value.isEmpty()) {
+                return null;
+            }
+            return value;
+        }
+    }
+
+    private ObjectMapper createMapperWithCustomDeserializer() {
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(String.class, new EmptyStringToNullDeserializer());
+
+        return JsonMapper.builder()
+                .addModule(module)
+                .defaultSetterInfo(JsonSetter.Value.forContentNulls(Nulls.FAIL))
+                .build();
+    }
+
+    @Test
+    public void nullsFailTest() {
```

Comment:
Same as with `Object[]`, no `null`s exists in test JSON; test does not trigger handling at all.